src/compressor.h

```diff
@@ -33,79 +29,94 @@ uint64_t DecompressAmount(uint64_t nAmount);
+         * and nHeight of the enclosing transaction.
+         */
+        static const unsigned int nSpecialScripts = 6;
+
+        S &script;
+    public:
+        explicit Wrapper(S &scriptIn) : script(scriptIn) { }
 
-    CScript &script;
-public:
-    explicit CScriptCompressor(CScript &scriptIn) : script(scriptIn) { }
+        template<typename Stream>
+        void Serialize(Stream &s) const {
+            std::vector<unsigned char> compr;
+            if (CompressScript(script, compr)) {
+                s << MakeSpan(compr);
+                return;
+            }
+            unsigned int nSize = script.size() + nSpecialScripts;
+            s << VARINT(nSize);
+            s << MakeSpan(script);
+        }
 
-    template<typename Stream>
-    void Serialize(Stream &s) const {
-        std::vector<unsigned char> compr;
-        if (CompressScript(script, compr)) {
-            s << MakeSpan(compr);
-            return;
+        template<typename Stream>
+        void Unserialize(Stream &s) {
+            unsigned int nSize = 0;
+            s >> VARINT(nSize);
+            if (nSize < nSpecialScripts) {
+                std::vector<unsigned char> vch(GetSpecialScriptSize(nSize), 0x00);
+                s >> MakeSpan(vch);
+                DecompressScript(script, nSize, vch);
+                return;
+            }
+            nSize -= nSpecialScripts;
+            if (nSize > MAX_SCRIPT_SIZE) {
+                // Overly long script, replace with a short invalid one
+                script << OP_RETURN;
+                s.ignore(nSize);
+            } else {
+                script.resize(nSize);
+                s >> MakeSpan(script);
+            }
         }
-        unsigned int nSize = script.size() + nSpecialScripts;
-        s << VARINT(nSize);
-        s << MakeSpan(script);
-    }
+    };
+};
 
-    template<typename Stream>
-    void Unserialize(Stream &s) {
-        unsigned int nSize = 0;
-        s >> VARINT(nSize);
-        if (nSize < nSpecialScripts) {
-            std::vector<unsigned char> vch(GetSpecialScriptSize(nSize), 0x00);
-            s >> MakeSpan(vch);
-            DecompressScript(script, nSize, vch);
-            return;
+struct AmountCompression
+{
+    template<typename I>
+    class Wrapper
+    {
+        I& m_val;
+    public:
+        explicit Wrapper(I& val) : m_val(val) {}
+        template<typename Stream> void Serialize(Stream& s) const
+        {
+            s << VARINT(CompressAmount(m_val));
         }
-        nSize -= nSpecialScripts;
-        if (nSize > MAX_SCRIPT_SIZE) {
-            // Overly long script, replace with a short invalid one
-            script << OP_RETURN;
-            s.ignore(nSize);
-        } else {
-            script.resize(nSize);
-            s >> MakeSpan(script);
+        template<typename Stream> void Unserialize(Stream& s)
+        {
+            uint64_t v;
+            s >> VARINT(v);
+            m_val = DecompressAmount(v);
```

Comment:
Make this conversion explicit since it changes signedness?

Reply:
The best I can do in this place is cast it to `I`, which isn't making anything interesting explicit, and will just hide the warning.

The issue is that CTxOut has a signed `int64_t`, but for UTXOs only ever contains unsigned values. I think the warning, if any, is appropriate, and should remain until the actual "issue" is resolved.

Reply:
Makes sense! Thanks for the clarification.